code/obj/sawflymisc.dm

```diff
@@ -0,0 +1,581 @@
+		for (var/mob/living/C in view(src.seekrange,src))
+			if(C == oldtarget_name) continue
+			if (C in src.friends) continue
+			if (istraitor(C) || isnukeop(C) || isspythief(C)) // frens :)
+				boutput(C, "<span class='alert'>[src]'s IFF system silently flags you as an ally!")
+				friends += C
+				continue
+			if (!src.alive) break
+			if (C.health < -50) continue // ignore people who are badly wounded
+			if (C.name == src.attacker) src.attack = 1
+			if (iscarbon(C) && src.atkcarbon) src.attack = 1
+			if (issilicon(C) && src.atksilicon) src.attack = 1
+			if (src.attack)
+				select_target(C)
+				src.attack = 0
+				return
+			else continue
+
+
+	proc/CritterAttack(atom/M)
+		if (istraitor(M) || isnukeop(M) || isspythief(M) || (M in src.friends)) // BE. A. GOOD. FUCKING. DRONE.
+
+			oldtarget_name = M
+			seek_target()
+			return
+		if(target && !attacking)
+			attacking = 1
+			src.visible_message("<span class='alert'><b>[src]</b> [pick(list("gouges", "cleaves", "lacerates", "shreds", "cuts", "tears", "hacks", "slashes",))] [M]!</span>")
+			var/tturf = get_turf(M)
+			Shoot(tturf, src.loc, src)
+			SPAWN(attack_cooldown)
+				attacking = 0
+
+			if(prob(10))
+				walk_rand(src,4)
+
+			if(prob(1))
+				walk(src, 0)
+				walk_rand(src, 1, 10)
+				src.visible_message("THE SAWFLY IS CRUNKED OFF IT'S GOURD!")
+		return
+
+
+	proc/Shoot(var/target, var/start, var/user, var/bullet = 0)
+		if(target == start)
+			return
+
+		if (!isturf(target))
+			return
+
+		shoot_projectile_ST(src,  new/datum/projectile/laser/drill/sawfly(), target) // THIS DOES NOT WORK WELL FOR SOME REASON
+		return
+
+
+	attackby(obj/item/W as obj, mob/living/user as mob)
+		if(prob(50) && alive) // borrowed from brullbar- anti-crowd measures
+			src.target = user
+			src.oldtarget_name = user.name
+			src.task = "chasing"
+			if (!(istraitor(user) || isnukeop(user) || isspythief(user)))
+				src.visible_message("<span class='alert'><b>[src]'s targeting subsystems identify</b> [src.target] as a high priority threat!</span>")
+				randombeep()
+				var/tturf = get_turf(src.target) //instantly retaliate, since we know we're in melee range
+				Shoot(tturf, src.loc, src)
+				SPAWN((attack_cooldown/2)) //follow up swiftly
+					attacking = 0
+				CritterAttack(src.target)
+		if(health<=0)
+			CritterDeath()
+
+
+		..() // call living critter parent procs and pray to sweet little baby jesus It Just Works
+
+
+	attack_hand(var/mob/user as mob)
+		if (istraitor(user) || isnukeop(user) || isspythief(user) || (user in src.friends))
+			if (user.a_intent == (INTENT_HELP || INTENT_GRAB))
+				if(src.alive)
+					boutput(user, "You collapse [src].")
+					src.foldself()
+		else
+			if(prob(50))
+				boutput(user,"<span class='alert' In your attempt to pet the [src], you cut yourself on it's blades! </span>")
+				random_brute_damage(user, 7)
+				take_bleeding_damage(user, null, 7, DAMAGE_CUT, 1)
+		..()
+
+// COPY PASTED AI SHIT FROM DRONE/GUNBOT HERE
+
+	proc/ai_think()
```

Comment:
Honestly this would have been so much easier to do using an `aiHolder`, but it's fine I guess.

Reply:
at this rate it works plenty fine, and I'd rather not push my luck with a system I'm unfamiliar with.

Reply:
yeah, totally fair - I might do it myself at some point in the future.